omnibus/package-scripts/agent/postinst

```diff
@@ -55,6 +55,27 @@ if [ "$DISTRIBUTION" != "Darwin" ]; then
             ;;
         esac
         #DEBHELPER#
+
+        # Set the installation information if not already present;
+        # This is done in posttrans for .rpm packages
+        if [ ! -f "$CONFIG_DIR/install_info" ]; then
+
+            if command -v dpkg >/dev/null 2>&1; then
+                tool=dpkg
+                tool_version=dpkg-$(dpkg --version | head -n 1 | cut -d " " -f 7 || echo "unknown")
```

Comment:
It got quite complex to get the version 🤦‍♀️ Getting the 7th fields seems fragile, since the wording of the output could change. Actually, I've checked the dpkg git history and there's one occurrence of this in 2006 where this changed from `Debian GNU/Linux package management program version x.x.x` to `Debian package management program version x.x.x`. Not very likely, but still feels hackish.

Also note `echo "unknown"` only applies to the last command (`cut ` in this case) so if `dpkg` or `head` fail we won't set the version to `unknown`. It's actually okay, since this prevents the command from failing and stopping the script, which is the most important part, but just so you know it doesn't do what seems obvious on a first read.

What do you think? We can still back up and not include the version here, or go with this and assume there is a (small) chance it returns the wrong version. As long as the script doesn't fail because of this, both work for me.

Reply:
> Getting the 7th fields seems fragile, since the wording of the output could change

I thought so too but I didn't think there was a working alternative (apart from maybe using `grep`?).
After [searching again](https://askubuntu.com/a/712202) it seems you could do:
```bash
dpkg-query --showformat='${Version}' --show dpkg
```

which works on the image I was using for testing, but I am not sure if we can assume `dpkg-query` is always available. I think it's preferable to using `cut` or `grep` in any case. Any thoughts?

 > Also note echo "unknown" only applies to the last command (cut in this case) so if dpkg or head fail we won't set the version to unknown. 

Oh, didn't know that, thank you for pointing it out! Should you wrap it in brackets as in `(dpkg --version | head -n 1 | cut -d " " -f 7) || echo "unknown"`?

Reply:
From my quick test, I think the parenthesis don't fix it either 🤷‍♀️ 

It looks like `dpkg-query` is installed with `dpkg` itself [1] so it does seem like a better solution to me. For RPM it already looks better IMO. To be extra safe, we can check `dpkg-query` is installed in the same `if` where we check that `dpkg` is installed. What do you think?

[1] https://packages.debian.org/sid/amd64/dpkg/filelist

Reply:
Okay, I will go with the extra check. And thank you for the link, I didn't know you could check the file list easily :)

Reply:
I discussed this in private with @julien-lebot (sorry again about missing your comment :sweat_smile:), I'll try to briefly summarize what I believe are the main points:

- An upside of the regex strategy is that `grep` is pretty standard and so it could work in more cases
- `dpkg --version` and `dpkg-query` can have different outputs; on an Ubuntu container the former outputs `1.19.0.5` while the latter outputs `1.19.0.5ubuntu2.3`. This higher specificity might lead to too many different versions and maybe make cardinality a concern. 
- It might also be the case that `dpkg --version` outputs some version like `1.19.0.5ubuntu2.3` and so the regex doesn't work.

Reply:
Agreed that having too many possible values is not ideal. What about `cut -d . -f 1-3`?

Reply:
No strong opinion on using a `regex` vs `dpkg-query`. It's safe to assume `grep` is always available, but it's also true that `dpkg-query` gives us the version without any additional text, so both should work.